sno/pk_generation.py

```diff
@@ -406,3 +453,45 @@ def feature_blobs(self):
             pk = self.decode_path_to_1pk(blob.name)
             if self.pk_filter(pk):
                 yield blob
+
+
+class FeaturePlusHashes:
+    _MORE_THAN_ONE_MATCH = object()
+
+    def __init__(self, feature, schema):
+        self.feature = feature
+        self.schema = schema
+        self.match = None
+
+    def __hash__(self):
+        return id(self)
+
+    def __eq__(self, other):
+        return other is self
+
+    def all_related_hashes(self):
+        value_hashes = []
+        feature_hash = 0
+        # Generate the overall hash of the feature, but in separate pieces.
+        for col in self.schema.non_pk_columns:
+            value = self.feature[col.name]
+            value_hash = hash(msg_pack([col.id, value]))
+            feature_hash ^= value_hash
+            value_hashes.append(value_hash)
+
+        # Add the feature at its overall hash.
+        yield feature_hash
+        # Remove each piece one at a time to make a hash for each variation of the feature missing exactly one field.
+        for value_hash in value_hashes:
+            yield feature_hash ^ value_hash
```

Comment:
oh - does this algorithm mean that these features have the same hash value?

```python
{'a': 1, 'b': 2, 'c': 3}
{'a': 2, 'b': 3, 'c': 1}
```


because it seems to be only hashing the values and not in an order-dependent way

Reply:
ah, ignore; I missed the `col.id` in `hash(msg_pack([col.id, value]))` above